### PR TITLE
Split TestCopier and TestMover

### PR DESCRIPTION
**Pull request by JinnyYi**
ref: https://github.com/beyondstorage/go-integration-test/issues/39

### CODE REVIEW

File: copier.go [hunk 1]
```diff
@@ -123,43 +123,127 @@ func TestCopier(t *testing.T, store types.Storager) {
...
-				 Convey("The error should be ErrObjectModeInvalid", func() {
-					 So(errors.Is(err, services.ErrObjectModeInvalid), ShouldBeTrue)
-				 })
-			 })
-		 }
 	})
 }
+
+func TestCopierWithDir(t *testing.T, store types.Storager) {
+	TestCopier(t, store)
```
Comment (Xuanwo) [inline copier.go]: How about let the service call the `TestCopier`?

File: copier.go [hunk 2]
```diff
@@ -123,43 +123,127 @@ func TestCopier(t *testing.T, store types.Storager) {
...
+
+	if d, ok := store.(types.Direr); ok {
```
Comment (Xuanwo) [inline copier.go]: The check is not needed, as it's a service duty.

File: copier.go [hunk 3]
```diff
@@ -123,43 +123,127 @@ func TestCopier(t *testing.T, store types.Storager) {
...
+				})
+			})
+		})
+	}
+}
+
+func TestCopierWithVirtualDir(t *testing.T, store types.Storager) {
+	TestCopier(t, store)
+
+	if d, ok := store.(types.Direr); ok {
```
Comment (Xuanwo) [inline copier.go]: ditto

File: copier.go (hunk 2 above)
Comment (JinnyYi) [reply copier.go]: Got it.

File: copier.go [hunk 4]
```diff
@@ -123,43 +123,123 @@ func TestCopier(t *testing.T, store types.Storager) {
...
 		})
+	})
+}
+
+func TestCopierWithDir(t *testing.T, store types.Storager) {
+	Convey("Given a basic Storager", t, func() {
+		c, ok := store.(types.Copier)
+		So(ok, ShouldBeTrue)
+
+		d, _ := store.(types.Direr)
```
Comment (Xuanwo) [inline copier.go]:
```suggestion
		d := store.(types.Direr)
```

It's OK to panic directly.

File: copier.go [hunk 5]
```diff
@@ -123,43 +123,123 @@ func TestCopier(t *testing.T, store types.Storager) {
...
+		})
+	})
+}
+
+func TestCopierWithVirtualDir(t *testing.T, store types.Storager) {
+	Convey("Given a basic Storager", t, func() {
+		c, ok := store.(types.Copier)
+		So(ok, ShouldBeTrue)
+
+		d, _ := store.(types.Direr)
```
Comment (Xuanwo) [inline copier.go]: ditto

**Pull request merged**